libs/cli/src/main/java/org/elasticsearch/cli/MultiCommand.java

```diff
@@ -45,6 +45,9 @@
      */
     public MultiCommand(final String description, final Runnable beforeMain) {
         super(description, beforeMain);
+        // Accepting -E here does not prevent subcommands receiving -E arguments, since we also set `posixlyCorrect` below.
+        // This stops option parsing when the first non-option is encountered.
+        parser.accepts("E", "Unused. Pass to a subcommand instead").withRequiredArg();
```

Comment:
I think this is a problem.
As you know, it's natural for people to expect that passing `-E` as the first argument to a multi-command will behave correctly. Currently they get an error, and have to work out what they've done wrong.

With this change they will no longer get an error, they will just get something that silently ignores their parameter. This is surprising and trappy behaviour.

If we want to do this, then I think we need to either:
1. make it an error to pass `-E` _and_ also call a subcommand (so `--help -E var.from.docker=true` works, but `-E var=true auto` does not); _OR_
2. actually pass the `-E` values into the subcommand like people expect.

Reply:
Unless I'm misunderstanding the behaviour here, and this already does (2), in which case can we make the comment more explicit?

Reply:
No, you're quite right. I was thinking primarily of the Docker case, but it would be much better if `-E` was respected wherever it is supplied. I'll look at this again.

Reply:
@tvernum I've addressed this in 31e4679 by passing along -E args to subcommands.